sql-ballerina/tests/call-procedures-test.bal

```diff
@@ -0,0 +1,393 @@
+// Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+//
+// WSO2 Inc. licenses this file to you under the Apache License,
+// Version 2.0 (the "License"); you may not use this file except
+// in compliance with the License.
+// You may obtain a copy of the License at
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+import ballerina/test;
+
+string proceduresDb = "procedures";
+string proceduresDB = urlPrefix + "9002/procedures";
+
+
+type StringDataForCall record {
+    string varchar_type;
+    string charmax_type;
+    string char_type;
+    string charactermax_type;
+    string character_type;
+    string nvarcharmax_type;
+};
+ 
+type StringDataSingle record {
+    string varchar_type;
+};
+
+
+
+@test:BeforeGroups {
+	value: ["procedures"]	
+} 
+function initproceduresContainer() {
+	initializeDockerContainer("sql-procedures", "procedures", "9002", "procedures", "call-procedures-test-data.sql");
+
+}
+
+@test:AfterGroups {
+	value: ["procedures"]	
+} 
+function cleanproceduresContainer() {
+	cleanDockerContainer("sql-procedures");
+}
+
+
+
+@test:Config {
+    groups: ["procedures"],
+    dependsOn: ["testCreateProcedures1"]
+}
+function testCallWithStringTypes() returns @tainted record {}|error? {
+    int id = 2;
+    MockClient dbClient = checkpanic new (url = proceduresDB, user = user, password = password);
+    ProcedureCallResult ret = checkpanic dbClient->call(`call InsertStringData(${id},'test1', 'test2     ', 'c', 'test3', 'd', 'test4');`);
+
+    ParameterizedQuery sqlQuery = `SELECT varchar_type,charmax_type, char_type, charactermax_type, character_type,
+                   nvarcharmax_type from StringTypes where id = ${id};`;
+
+    stream<record{}, Error> streamData = dbClient->query(sqlQuery,StringDataForCall);
+    stream<StringDataForCall,Error> queryData = <stream<StringDataForCall,Error>>streamData;
+    StringDataForCall? returnData = ();
+    error? e = queryData.forEach(function(StringDataForCall data) {
+        returnData = data;
```

Comment:
Is there an only one record select here? Else this code will override the `returnData` variable till the last record

Reply:
yes, only one record will select from that method

Reply:
In these special cases, it is better to add a comment explaining why it is overridden